annotation/tests/test_tasks_services.py

```diff
@@ -295,3 +302,166 @@ def test_validate_user_actions_invalid_states(
             validation_user=True,
         )
     assert re.match(expected_error_message_pattern, excinfo.value.detail)
+
+
+def test_create_annotation_task():
+    with patch("sqlalchemy.orm.Session", spec=True) as mock_session:
+
+        db_session = mock_session()
+        create_annotation_task(
+            db_session,
+            ManualAnnotationTaskInSchema(
+                file_id=1,
+                pages={1, 2},
+                job_id=2,
+                user_id=uuid4(),
+                is_validation=True,
+                deadline=None,
+            ),
+        )
+
+        assert db_session.add.call_count == 2
+        db_session.commit.assert_called_once()
+
+
+@pytest.fixture
+def mock_db_session():
+    with patch("sqlalchemy.orm.Session", spec=True) as mock_session:
+        yield mock_session()
+
+
+def test_read_annotation_tasks_with_file_and_job_ids(mock_db_session):
+
+    mock_query = mock_db_session.query.return_value
+    mock_query.filter_by.return_value = mock_query
+    mock_query.filter.return_value = mock_query
+    mock_query.count.return_value = 1
+    mock_query.limit.return_value.offset.return_value.all.return_value = [
+        "task1"
+    ]
+
+    total_objects, annotation_tasks = read_annotation_tasks(
+        db=mock_db_session,
+        search_params={"file_ids": [1, 2], "job_ids": [3]},
+        pagination_page_size=10,
+        pagination_start_page=1,
+        tenant="example_tenant",
+    )
+
+    assert total_objects == 1
+    assert annotation_tasks == ["task1"]
+
+    mock_query.filter_by.assert_called_once()
+    mock_query.limit.assert_called_once_with(10)
+
+
+@pytest.mark.parametrize(
+    ("search_id", "search_name", "ids_with_names", "expected_result"),
+    (
+        (1, "Task 1", {1: "Task 1", 2: "Task 2"}, ([1], {1: "Task 1"})),
+        (1, None, {1: "Task 1", 2: "Task 2"}, ([1], {})),
+    ),
+)
+def test_validate_ids_and_names_valid_cases(
+    search_id, search_name, ids_with_names, expected_result
+):
+    result = validate_ids_and_names(
+        search_id=search_id,
+        search_name=search_name,
+        ids_with_names=ids_with_names,
+    )
+    assert result == expected_result
+
+
+@pytest.mark.parametrize(
+    ("search_id", "search_name", "ids_with_names"),
+    [
+        (3, "Task 3", {1: "Task 1", 2: "Task 2"}),
+        (1, "Task 1", {}),
+        (None, "Task 2", None),
+    ],
+)
+def test_validate_ids_and_names_raises_error(
+    search_id, search_name, ids_with_names
+):
+    with pytest.raises(HTTPException) as exc_info:
+        validate_ids_and_names(
+            search_id=search_id,
+            search_name=search_name,
+            ids_with_names=ids_with_names,
+        )
+    assert exc_info.value.status_code == 404
+
+
+@pytest.mark.parametrize(
```

Comment:
this parametrize is overwhelming, let's make it simpler. Could you split it into multiple tests or simplify input data in parametrize?